single-line-diagram-force-layout/src/main/java/com/powsybl/sld/force/layout/ForceSubstationLayout.java

```diff
@@ -77,53 +72,59 @@ public ForceSubstationLayout(SubstationGraph substationGraph,
+    private Graph<Point, Spring> toJgrapht(SubstationGraph graph) {
+        Graph<Point, Spring> pseudograph = new Pseudograph<>(Spring.class);
+
+        // Create nodes
+        for (VoltageLevelGraph voltageLevelGraph : graph.getNodes()) {
+            String id = voltageLevelGraph.getVoltageLevelInfos().getId();
+            Point point = new Point(id, random.nextDouble() * 1000, random.nextDouble() * 1000);
+            pseudograph.addVertex(point);
         }
 
-        for (Node multiNode : getGraph().getMultiTermNodes()) {
+        // Create edges
+        for (Node multiNode : graph.getMultiTermNodes()) {
             List<Node> adjacentNodes = multiNode.getAdjacentNodes();
-            List<Edge> adjacentEdges = multiNode.getAdjacentEdges();
-            NodeImpl node1 = (NodeImpl) undirectedGraph.getNode(adjacentNodes.get(0).getGraph().getVoltageLevelInfos().getId());
-            NodeImpl node2 = (NodeImpl) undirectedGraph.getNode(adjacentNodes.get(1).getGraph().getVoltageLevelInfos().getId());
-            undirectedGraph.addEdge(new EdgeImpl(adjacentEdges.get(0).toString() + "_" + adjacentEdges.get(1).toString(), node1, node2, 0, 1, false));
+
+            Point point1 = getPointAtIndex(pseudograph, adjacentNodes, 0);
+            Point point2 = getPointAtIndex(pseudograph, adjacentNodes, 1);
+            pseudograph.addEdge(point1, point2);
 
             if (adjacentNodes.size() == 3) {
-                NodeImpl node3 = (NodeImpl) undirectedGraph.getNode(adjacentNodes.get(2).getGraph().getVoltageLevelInfos().getId());
-                undirectedGraph.addEdge(new EdgeImpl(adjacentEdges.get(1).toString() + "_" + adjacentEdges.get(2).toString(), node2, node3, 0, 1, false));
+                Point point3 = getPointAtIndex(pseudograph, adjacentNodes, 2);
+                pseudograph.addEdge(point2, point3);
             }
         }
 
-        // Creating the ForceAtlas and run the algorithm
-        ForceAtlas2 forceAtlas2 = new ForceAtlas2Builder().buildLayout();
-        forceAtlas2.setGraphModel(graphModel);
-        forceAtlas2.resetPropertiesValues();
-        forceAtlas2.setAdjustSizes(true);
-        forceAtlas2.setOutboundAttractionDistribution(false);
-        forceAtlas2.setEdgeWeightInfluence(1.5d);
-        forceAtlas2.setGravity(10d);
-        forceAtlas2.setJitterTolerance(.02);
-        forceAtlas2.setScalingRatio(15.0);
-        forceAtlas2.initAlgo();
-        int maxSteps = 1000;
-
-        for (int i = 0; i < maxSteps && forceAtlas2.canAlgo(); i++) {
-            forceAtlas2.goAlgo();
-        }
-        forceAtlas2.endAlgo();
+        return pseudograph;
+    }
+
+    private Point getPointAtIndex(Graph<Point, Spring> graph, List<Node> adjacentNodes, int index)  {
+        String adjacentNodeId1 = adjacentNodes.get(index).getGraph().getVoltageLevelInfos().getId();
+        return (Point) graph.vertexSet().stream().filter(p -> p.getId().equals(adjacentNodeId1)).findAny().get();
```

Comment:
Beware, you might be filtering out a big vertex set here, as this can be called many many times. I think we can do better here, but let's see that later, once we use generic graphs.